coredns/resolver/service_info.go

```diff
@@ -88,3 +88,7 @@ func (si *serviceInfo) selectIP(checkCluster func(string) bool) *DNSRecord {
 
 	return nil
 }
+
+func (si *serviceInfo) isHeadless() bool {
```

Comment:
QQ: I know it's a small function, but performance-wise -  isn't it better to keep the current code ? I mean calculate it once when serviceInfo is created and not every time DNS resolver is called.

Reply:
I made it a method for readability and to avoid duplicate storage. It might be a little faster as a variable but the difference would be extremely negligible (probably in nanosec). Plus compilers are pretty smart about optimizing and caching frequently accessed code.